source/codegen/metadata/nirfmxinstr/functions.py

```diff
@@ -1720,10 +1738,12 @@
             },
             {
                 'direction': 'out',
+                'enum': 'Boolean',
                 'name': 'selfCalibrateValid',
                 'type': 'int32'
             },
             {
+                'bitfield_as_enum_array': 'SelfCalStep',
```

Comment:
Should we be marking the output parameters using the `Personality` enum with this tag? `Personality` also looks like a bitfield enum but I'm not sure if there's some nuances between the methods returning it vs. this one with `SelfCalStep`.

Reply:
Good question. `Personality` is a bitfield, but most functions that accept it are just passing or returning a single value (or so it seems).

There is one exception: `personalityFilter`. That's definitely a bitfield input.

`bitfield_as_enum_array` currently only works for outputs. That seems surprising. But the reason for that is that using regular-old-enums for bitfield *inputs* actually works OK. In most client languages (i.e., python, C++) you can `or` together the enum values and pass them for the `_raw` field and you get a pretty reasonable experience. We've been doing that since the MI services. 

Trying to do that with outputs is ... less obvious. We'd have to always return the raw field and would never know how to populate the non-raw. And the client would have to do bitwise logic on the result with enum values that don't actually match the field. So, I added `bitfield_as_enum_array` the first time I hit a bitfield output.

I was still thinking about adding a task to use `bitfield_as_enum_array` for `personalityFilter`, which would mean adding support for `bitfield_as_enum_array` inputs. That should be a better experience with a possible exception for passing a single value (still not bad). But it is a little extra work, and we have already decided (multiple times) that regular enums were "good enough for this case.

Reply:
Ah okay, that all makes sense. Looks good then.

And yeah, I think it's fine to keep it as is with the "regular enums are good enough for input bit fields" rationale. Good to keep in mind, though, if we do ever decide the improved clarity for clients filling out input fields is worth it.